### ISSUE TITLE
Deactivate users after [just over] a year

### ISSUE THREAD

**Issue by AmaniKrayemRCPCH** [rcpch/national-paediatric-diabetes-audit#406]
*EDIT @mbarton* - refocusing this issue based on the discussion below. Automatically make users inactive if they haven't logged in for a year. Inactive users can't log in. An RCPCH Audit Team member must then activate their account before the can log in.

We should force a password reset too as part of this process, although that may be done automatically by Django's password expiry mechanism?

Feedback from NPDA meeting 10/07/25 - the expiry time should be more than year to cover off people who only log in just before the deadline in each audit year and also anyone who goes on maternity leave + accrued annual leave

**Comment by mnmnk43434** [comment]
Hello, we tried to solve the issue.

This is what we did:

Implement a method to delete inactive users after 2 years and create a management command to run this check periodically.

You can review changes in this commit: https://github.com/mnmnk43434/rcpch-national-paediatric-diabetes-audit-406/commit/994d4f205bd8e69aca68d6d285b4d2ae5a00043f.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by eatyourpeas** [comment]
Thanks @AmaniKrayemRCPCH sure we can do something like this. My main worry is that you can't really delete users because they are (and should be) forever tied to the data they entered and the decisions they made. We should have a paper trail even 10 years from now of who did what when. But rather than delete users, we can make them inactive, so that they do not appear in our lists of active users. If for some reason they become involved again in future we can simply reactivate their account.

**Comment by mbarton** [comment]
Agreed we can't delete users but we may need to anonymise their personal data after a given amount of time.

I will check with the RCPCH information governance team what our obligations are as they will flow down from our contract with HQIP

**Comment by mbarton** [comment]
Their guidance is that deactivating accounts (and requiring an admin to reactivate them before they can login) is good security practice and it would be normal to remove the PII associated with them after a given amount of time. We can decide what that period is, they suggested 6 months to a year but I think for some audits we'd want to go longer as people might rush to fill them in before a yearly deadline.

For actually implementing it, they said given our restriction on not deleting users it's low risk and we could argue not worth the effort to introduce anonymising the PII. So let's leave this in the backlog pending any further discussions.
